demo/src/routes/menu/Versions.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+	import {get, writable} from '@amadeus-it-group/tansu';
+	import type {ReadableSignal, Writable} from '@amadeus-it-group/tansu';
+	import type {Directive} from '@agnos-ui/svelte/types';
+	import {createHasFocus} from '@agnos-ui/svelte/services/focustrack';
+	import {onMount} from 'svelte';
+	import {page} from '$app/stores';
+	import {pathToRoot$, selectedFramework$} from '$lib/stores';
+
+	interface Version {
+		folder: string;
+		version: string;
+	}
+
+	const versionsParsed: Version[] = [{folder: 'latest', version: 'Latest'}];
+
+	onMount(() => {
+		fetch('https://amadeusitgroup.github.io/AgnosUI/versions.json').then((res) => {
+			res.json().then((result) => {
+				versionsParsed.push(...result);
+			});
+		});
+		({hasFocus$, directive} = createHasFocus());
+	});
+
+	const currentVersion$: Writable<Version> = writable(versionsParsed[0]);
+	const open$ = writable(false);
+
+	let hasFocus$: ReadableSignal<boolean>;
+	let directive: Directive;
+	$: $open$ = $hasFocus$;
+
+	function giveFocus(el: HTMLAnchorElement, index: number) {
+		if (index === 0) {
+			el.focus();
+		}
+	}
+
+	function getRedirectionURL(version: string) {
+		const currentURL = get(page).url;
+		if (get(page).data['since'] <= version) {
+			return currentURL.toString().replace($currentVersion$.folder, version);
+		} else {
+			return `${$pathToRoot$.replace($currentVersion$.folder, version)}docs/${$selectedFramework$}/getting-started/introduction`;
+		}
+	}
+</script>
+
+<div class="nav-item ms-3">
+	<div class="dropdown">
+		<button
+			class="btn nav-link dropdown-toggle align-items-center d-flex"
```

Comment:
If you add the PREVIEW, targetting main, it could be good to change the styling of the button to make it clear to users it's not released yet.
Maybe something like this ?
```svelte
<button
  class="btn nav-link dropdown-toggle align-items-center d-flex {$currentVersion$.version === 'PREVIEW' ? 'badge text-bg-warning' : ''}"
 ></button>
```